src/services/v1_service/service.ts

```diff
@@ -1,8 +1,24 @@
+import {readFileSync} from "fs";
+import {resolve} from "path";
 import {Errors, GET, Path, QueryParam} from "typescript-rest";
-import {AnnotationStatus, Aspect, makeGroupedAnnotations, readData,} from "../../utils/ingest";
+import {
+  AnnotationStatus,
+  Aspect,
+  ingestData,
+  makeAnnotationIndex, StructuredData,
+  UnstructuredText
+} from "../../utils/ingest";
 import {queryAnnotated, QueryOption, Segment, Strategy} from "../../queries/queries";
 
-const { geneMap, annotations, groupedAnnotations } = readData();
+// TODO use data fetcher rather than files.
+console.log("Begin reading data");
+const genesText = readFileSync(resolve("src/assets/gene-types.txt")).toString();
+const annotationsText = readFileSync(resolve("src/assets/gene_association.tair")).toString();
+const unstructuredText: UnstructuredText = { genesText, annotationsText };
+const maybeDataset = ingestData(unstructuredText);
+if (!maybeDataset) throw new Error("failed to parse data");
+const dataset: StructuredData = maybeDataset;
+console.log("Finished parsing data");
```

Comment:
@scourgemancer once this gets merged, this is going to be the only section you should need to replace. Instead of us hardcoding a file-read for the two files, you'd instantiate your fetcher-scheduler thing.